### DISCUSSION TITLE
Custom PDK

### DISCUSSION THREAD

**Discussion by ashkanr65** [The-OpenROAD-Project/OpenLane#2068] [General]
Hi, 
I want to make a custom PDK for OpenLane. I have a techfile, gds files of cell libraries, LEF files, and a liberty file. But I don’t know where I should put them, how to address them, and how to make OpenLane use those files. Is there any documentation? Or has somebody done that?

**Comment by kareefardi** [comment]
Check these out and let us know if you have any questions:
- https://openlane.readthedocs.io/en/latest/for_developers/pdk_structure.html
- https://openlane.readthedocs.io/en/latest/reference/pdk_configuration.html

**Reply by ashkanr65** [reply]
Hi,

I added all folders similar to the structures and config. However, when I ran an example (spm model with cconfig changed), I received this error:

[ERROR]: Failed to compare PDKs.
/home/ashkan/OpenLane/pdks/sky130A not found.
Traceback (most recent call last):
File "/openlane/dependencies/verify_versions.py", line 73, in verify_versions
raise Exception(f"{pdk_dir} not found.")
Exception: /home/ashkan/OpenLane/pdks/sky130A not found.

[ERROR]: Please update your environment. OpenLane will now quit.

**Reply by kareefardi** [reply]
Did you overwrite `PDK_ROOT` and `PDK` env variables ?
